src/components/tabs/Best/BestTimer.tsx

```diff
@@ -0,0 +1,115 @@
+import { type FC, useEffect, useState } from "react";
+import "react-loading-skeleton/dist/skeleton.css";
+import { toast } from "react-hot-toast";
+
+type BestTimerProps = {
+  timeInMs: number;
+  isRunning: boolean;
+  isLoading: boolean;
+  toggleTimerHandler: () => void;
+};
+
+const BestTimer: FC<BestTimerProps> = ({
+  timeInMs,
+  isRunning,
+  isLoading,
+  toggleTimerHandler,
+}) => {
+  const [showProgress, setShowProgress] = useState(true);
+  const [clickEffect, setClickEffect] = useState(false);
+
+  useEffect(() => {
+    const lastVisit = localStorage.getItem("lastVisit");
+    if (lastVisit === null) {
+      toast("Tap the timer to hide it", {
+        duration: 5000,
+        position: "top-center",
+      });
+    } else {
+      if (
+        new Date().valueOf() - Date.parse(lastVisit).valueOf() >
```

Comment:
nit: might want to abstract this out to a helper function to improve readability